src/uint/modular/constant_mod.rs

```diff
@@ -99,6 +109,23 @@ impl<MOD: ResidueParams<LIMBS>, const LIMBS: usize> Residue<MOD, LIMBS> {
         }
     }
 
+    /// Instantiates a new `Residue` that represents this `integer` mod `MOD` if the modulus is odd.
+    /// Returns a `CtOption` that is `None` if the provided modulus is not odd; this is a safer version of [`new`][`Residue::new`], which can panic.
+    pub fn new_checked(integer: &Uint<LIMBS>) -> CtOption<Self> {
```

Comment:
I think the same thing as in the DynResidue PR is applicable - can we have a single constructor to avoid repeating the logic?

Reply:
Sure can!